File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/Remote.cs
namespace BridgeDesignPattern.Example1;

public class Remote
{
	protected readonly IDevice _device;

	public Remote(IDevice device)
	{
		_device = device;
	}

	public void TogglePower()
	{
		if (_device.IsEnabled)
			_device.Disable();

		else _device.Enable();
	}

	public void VolumeUp()
	{
		if (_device.Volume < 100)
			_device.Volume++;
	}

	public void VolumeDown()
	{
		if (_device.Volume > 0)
			_device.Volume--;
	}

	public void ChannelUp()
	{
		_device.Channel++;
	}

	public void ChannelDown()
	{
		if (_device.Channel > 1)
			_device.Channel--;
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/src/Remote.ts
import { IDevice } from './IDevice';

export class Remote {
  constructor(protected readonly device: IDevice) {}

  togglePower(): void {
    if (this.device.isEnabled) this.device.disable();
    else this.device.enable();
  }

  volumeUp(): void { if (this.device.volume < 100) this.device.volume++; }
  volumeDown(): void { if (this.device.volume > 0) this.device.volume--; }
  channelUp(): void { this.device.channel++; }
  channelDown(): void { if (this.device.channel > 1) this.device.channel--; }
}
